pkg/3scale/amp/operator/base_apimanager_logic_reconciler.go

```diff
@@ -90,11 +92,16 @@ func (r *BaseAPIManagerLogicReconciler) ReconcileGrafanaDashboard(desired *grafa
 		return err
 	}
 
-	if !kindExists {
-		r.Logger().Info("Install grafana-operator in your cluster to create grafanadashboards objects", "Error creating grafanadashboard object", desired.Name)
+	if !r.apiManager.IsMonitoringEnabled() && !kindExists {
 		return nil
 	}
 
+	if r.apiManager.IsMonitoringEnabled() && !kindExists {
+		err = fmt.Errorf("Error creating grafanadashboard object '%s'", desired.Name)
+		r.Logger().Error(err, "Install grafana-operator in your cluster to create grafanadashboards objects")
+		return err
```

Comment:
Why return error? with this implementation 3scale will not be deployed when monitoring enabled and CRDs not available.

Reply:
See my https://github.com/3scale/3scale-operator/pull/509#discussion_r515076719 comment.

I was commenting exactly this.

If we don't return an error when monitoring is enabled it will proceed with creating other elements, and it will mask that something's gone wrong
If we return an error it will not proceed creating other elements. 3scale will not be deployed but that something's gone wrong will be more evident.

It's up to us what we want to do with this behavior.

Reply:
I would not change the current behavior. Any reason to not deploy 3scale if monitoring stack is not available?

Reply:
It seems just a conceptual behavior.

"If a user sets monitoring:enabled, and the kind does not exist, is it correct what we don't return an error"?

Reply:
We continue with current behavior of logging and proceeding with other elements.

I'll add an event too to try to make it more clear

Reply:
We decide to keep current behavior. `nil` is returned to continue deploying other objects.
We now log Error instead of Info.
Additionally we now send event (of type Warning, there's no Error event type)